algorithms/bc/agent.py

```diff
@@ -99,52 +95,60 @@ def __init__(self, env, args, device):
             self.load_params(args.load_from)
 
         # noise instance to make randomness of action
-        self.noise = OUNoise(action_dim, self.args.seed, theta=0.0, sigma=0.0)
+        self.noise = OUNoise(self.action_dim, self.args.seed, theta=0.0, sigma=0.0)
 
         # replay memory
         self.memory = ReplayBuffer(
             hyper_params["BUFFER_SIZE"],
             hyper_params["BATCH_SIZE"],
             self.args.seed,
-            self.device,
+            device,
         )
 
         # load demo replay memory
-        with open(hyper_params["DEMO_PATH"], "rb") as f:
+        with open(DEMO_PATH, "rb") as f:
             demo = pickle.load(f)
 
         self.demo_memory = ReplayBuffer(
-            len(demo),
-            hyper_params["DEMO_BATCH_SIZE"],
-            self.args.seed,
-            self.device,
-            demo,
+            len(demo), hyper_params["DEMO_BATCH_SIZE"], self.args.seed, device, demo
         )
 
-    def select_action(self, state):
+    def select_action(self, state: np.ndarray) -> torch.Tensor:
         """Select an action from the input space."""
-        selected_action = self.actor_local(state)
-        selected_action += torch.tensor(self.noise.sample()).float().to(self.device)
+        state = torch.FloatTensor(state).to(device)
+        selected_action = self.actor(state)
+        selected_action += torch.tensor(self.noise.sample()).float().to(device)
 
-        action_low = float(self.env.action_space.low[0])
-        action_high = float(self.env.action_space.high[0])
+        action_low = self.action_low
+        action_high = self.action_high
         selected_action = torch.clamp(selected_action, action_low, action_high)
 
         return selected_action
 
-    def step(self, state, action):
+    def step(
+        self, state: np.ndarray, action: torch.Tensor
+    ) -> Tuple[np.ndarray, np.float64, bool]:
         """Take an action and return the response of the env."""
-        action = action.detach().to("cpu").numpy()
+        action = action.detach().cpu().numpy()
         next_state, reward, done, _ = self.env.step(action)
         self.memory.add(state, action, reward, next_state, done)
 
         return next_state, reward, done
 
-    def update_model(self, experiences, demo):
+    def update_model(
+        self,
+        experiences: Tuple[
+            torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor
+        ],
+        demos: Tuple[
+            torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor
+        ],
```

Comment:
As I checked, type of demos is Tuple, too. [ref](https://github.com/medipixel/reinforcement_learning_examples/blob/ac37c00ecc404acb5c129fc307a139341f26e66c/algorithms/bc/agent.py#L268)
So I modify this line.